// hltv/lib/config.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultConfig = exports.defaultLoadPage = void 0;
var https_1 = require("https");
var request = __importStar(require("request"));
var random_useragent_1 = __importDefault(require("random-useragent"));
var defaultLoadPage = function (httpAgent) { return function (url) {
    return new Promise(function (resolve) {
        request.get(url, {
            gzip: true,
            agent: httpAgent,
            headers: {
                'User-Agent': random_useragent_1.default.getRandom(function (ue) {
                    return [
                        '/Browsers - Windows',
                        '/Browsers - Linux',
                        '/Browsers - Mac'
                    ].includes(ue.folder);
                })
            }
        }, function (err, __, body) {
            if (err) {
                throw err;
            }
            resolve(body);
        });
    });
}; };
exports.defaultLoadPage = defaultLoadPage;
var defaultAgent = new https_1.Agent();
exports.defaultConfig = {
    httpAgent: defaultAgent,
    loadPage: exports.defaultLoadPage(defaultAgent)
};
